### PR TITLE
Fix/sidebar and category numbers

### PR DESCRIPTION
**Pull request by Jacksonmcguire**
## What does this PR do?

- Add a no Api's available error message when there are no search results, add available options for the category buttons with a dynamic function(though not fully dynamic yet due to some render bugs). Add favorite cards to the saved Apis sidebar, do some minor styling changes to the sidebar. 

## How should it be tested in the UI?

- Fetching down the branch to see if the favorite functionality works as expected, and the category buttons hold their initial possible options values.

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -45,53 +46,13 @@ class App extends React.Component<Props> {
...
-    matchedCards = this.filterByCategory(matchedCards, stateObj)
-    matchedCards = this.filterBySearch(matchedCards, stateObj)
-    matchedCards = this.filterByAuth(matchedCards, stateObj)
-    matchedCards = this.filterByHTTPS(matchedCards, stateObj)
-    matchedCards = this.filterByCors(matchedCards, stateObj)
+    matchedCards = filterByCategory(matchedCards, stateObj)
+    matchedCards = filterBySearch(matchedCards, stateObj)
+    matchedCards = filterByAuth(matchedCards, stateObj)
+    matchedCards = filterByHTTPS(matchedCards, stateObj)
+    matchedCards = filterByCors(matchedCards, stateObj)
```
Comment (itsnameissteven) [inline src/components/App/App.tsx:55]: I like this. nice and clean

File: src/components/FilterForm/FilterForm.tsx [hunk 2]
```diff
@@ -34,13 +34,16 @@ export class FilterForm extends React.Component<FilterProps> {
     }
   }
 
-  handleChange = (e: any):void => {
+
+  handleChange = (e: React.ChangeEvent<HTMLSelectElement> | React.ChangeEvent<HTMLInputElement>):void => {
```
Comment (itsnameissteven) [inline src/components/FilterForm/FilterForm.tsx:38]: thanks for changing this from "any"

**Review by itsnameissteven — APPROVED** [review]
looks good to me!

File: src/filterMethods.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }
+}
+
+export const filterByCors = (matchedCards: Api[], stateObj: FilterState) => {
+  if (stateObj.Cors !== 'all') {
+    return matchedCards.filter(api => api.Cors === stateObj.Cors)
+  } else {
+    return matchedCards
+  }
+}
```
Comment (cbmackintosh) [inline src/filterMethods.ts:42]: Great idea moving these filter functions into a different file. Keeps the codebase more modular. App component was becoming a real behemoth.

**Review by cbmackintosh — APPROVED** [review]
Nice work. The category lengths are a great added feature. UI is coming together nicely

**Pull request merged**